pg_shard.c

```diff
@@ -378,6 +378,25 @@ ErrorIfQueryNotSupported(Query *queryTree)
 						errmsg("unsupported utility statement")));
 	}
 
+	/*
+	 * Reject subqueries which are in SELECT or WHERE clause.
+	 * Queries which include subqueries in FROM clauses are rejected below.
+	 */
+	if (queryTree->hasSubLinks == true)
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("cannot perform distributed planning on this query"),
+						errdetail("Subqueries are currently unsupported.")));
```

Comment:
I'd like to rewrite all of these details as _FEATURE(S) is/are not supported in distributed queries._. So here we'd have: _Subqueries are not supported in distributed queries._.